src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -614,10 +615,20 @@ private async Task<UserDecryptionOptions> CreateUserDecryptionOptionsAsync(User
                 encryptedPrivateKey = device.EncryptedPrivateKey;
                 encryptedUserKey = device.EncryptedUserKey;
             }
+
+            var allDevices = await _deviceRepository.GetManyByUserIdAsync(user.Id);
+            // Checks if the current user has any devices that are capable of approving login with device requests except for
```

Comment:
Can we add a `note: this doesn't check for if the users have configured the devices to be capable of approving requests as that is a client side setting.`?